darts/tests/models/forecasting/test_tsmixer.py

```diff
@@ -362,3 +362,25 @@ def test_time_batch_norm_2d_gradients(self):
         output.mean().backward()
 
         assert input_tensor.grad is not None
+
+    @pytest.mark.parametrize("project_first_layer", [True, False])
+    def test_project_first(self, project_first_layer):
+        ts = tg.sine_timeseries(length=36, freq="h")
+        input_len = 12
+        output_len = 6
+
+        model = TSMixerModel(
+            input_chunk_length=input_len,
+            output_chunk_length=output_len,
+            n_epochs=1,
+            project_first_layer=project_first_layer,
+            # Cover case of projecting future covs back to input dims
+            add_encoders={"cyclic": {"future": "hour"}},
+            **tfm_kwargs,
+        )
+        model.fit(ts)
+
+        if project_first_layer:
+            assert model.model.sequence_length == output_len
+        else:
+            assert model.model.sequence_length == input_len
```

Comment:
Can the test also include a call to `predict()` to make sure it works as well (even if the forward pass is already occurring in the call to `fit()`)?